app/grandchallenge/evaluation/models.py

```diff
@@ -1715,6 +1730,42 @@ def get_absolute_url(self):
             },
         )
 
+    @cached_property
+    def has_matching_algorithm_interfaces(self):
+        return sorted(
+            [int.pk for int in self.phase.algorithm_interfaces.all()]
+        ) == sorted(
+            [int.pk for int in self.algorithm_image.algorithm.interfaces.all()]
+        )
+
+    @cached_property
+    def has_blocking_algorithm_jobs(self):
+        # are there non-successful jobs for the algorithm image, model and archive?
+        algorithm_interfaces = self.phase.algorithm_interfaces.all()
+
+        items = get_archive_items_for_interfaces(
+            algorithm_interfaces=algorithm_interfaces,
+            archive_items=self.phase.archive.items.all(),
+        )
+        non_success_statuses = [
+            status[0]
+            for status in Job.STATUS_CHOICES
+            if status[0] != Job.SUCCESS
+        ]
+
+        jobs = get_valid_jobs_for_interfaces_and_archive_items(
```

Comment:
I'm a little bit worried that this will take too long for a synchronous view. Could we short-cut it? Something like:

```python
    @cached_property
    def has_blocking_algorithm_jobs(self):
        return Job.objects.exclude(status=Job.SUCCESS).filter(inputs__archive_items__archive__phase=self.phase, algorithm_image=self.algorithm_image, algorithm_model=self.algorithm_model).exists()
```

This way you wouldn't need to modify `get_valid_jobs_for_interfaces_and_archive_items`.

Reply:
This would match jobs with partial inputs (or additional inputs). We only want to know about non-successful jobs with the exact same inputs as the archive items, so I'm afraid the complex check is necessary.

Reply:
I get your point about this possibly taking too long though. Is this something we could compute periodically and save on the evaluation object? I think it shouldn't change, technically, after an evaluation finishes, but there might be corner cases.

Reply:
Hmm. We shouldn't have a periodic task as the data would then not be in sync. Why not do this asynchronously and show the evaluation as failed? It is anyway not possible to evaluate in this scenario.

Reply:
That could work.